docs/architecture/domain.md

```diff
@@ -0,0 +1,116 @@
+## Definition Model
+
+![Definition](assets/definition-model.png)
+
+### Pipeline
+
+A pipeline represents a collection of [jobs] that share the same source code. These jobs are executed in the order defined by the [workflow]. Only one job is required to be defined in every pipeline:
+
+ - `main` - Job that runs on each change to the source code.  _Note: this is also the same job that runs on proposed changes via GitHub Pull Requests_
+
+### Workflow
+
+Workflow is the defined order that [jobs] will execute in. Jobs can be executed in parallel, series, and a combination of the two to allow for all possibilities. All jobs executed in a given workflow run share:
+
+ - Source code checked out to the same git commit from the `main` [build] that triggered this
+ - Access to the [metadata] from the `main` [build] that triggered this
+
+There are two requirements for your pipeline's workflow:
+
+ - Must start with `main` (triggered when the git repository changes)
+ - Must contain every job (no non-triggered jobs)
+
+In the following example, this is the flow that will happen:
+```yaml
+workflow:
+    - main
+    - publish
+    - parallel:
+        - series:
+            - deploy-west
+            - validate-west
+        - series:
+            - deploy-east
+            - validate-east
+```
+
+ - `main` triggers `publish`
+ - `publish` triggers `deploy-west` and `deploy-east` in parallel
+ - `deploy-west` triggers `validate-west`
+ - `deploy-east` triggers `validate-east`
+
+### Job
+
+A job consists of executing multiple sequential [steps] inside a specified [container]. If any step in the series fails, then the entire job is considered failed and subsequent steps will be skipped (unless configured otherwise).
+
+Jobs share four pieces of context between steps:
+
+ - Source Code - Checked out to a given revision at the start of the job
+ - Filesystem - Any changes made by a step to the filesystem will be visible to subsequent steps
```

Comment:
You already mention that Jobs share these things between steps, so no need to add the `will be visible to subsequent steps`